### ISSUE TITLE
Installing from GIT and the AUR Corrupts package

### ISSUE THREAD

**Issue by ghost** [Jazqa/kwin-quarter-tiling#14]
I installed the package from GIT, but then thought to check out the AUR package for easier updating (The current update process can be a little iffy). This has now broken the script, and when clicking the configure button under Kwin Scripts dialog, it shows 2 identical tabs.

There is an install.sh script included with this, but perhaps a way to completely purge the script would help too? By doing this we can then more easily diagnose issues as well (update vs fresh install)

In the interim, how would I purge this script?

**Comment by Jazqa** [comment]
AUR version installs itself to /usr/share/kwin/scripts instead of .local. Just remove it from there.

Also, any ideas for the updating process are appreciated. I'm currently working on the script daily, so I'm updating it too much for a versioned release on the releases-page and KDE Store is having login issues (I've e-mailed the admin). Right now git fetch && pull is the quickest solution I can think of (being one comman).

**Comment by ghost** [comment]
So I've managed to do that, and went ahead and removed the old script and re-downloaded via GIT etc. I cannot for the life of my to get the script to work now.

What tools do you use to debug it? None of the key combos i setup work, nor do the defaults that you've setup.

**Comment by Jazqa** [comment]
If the AUR build screwed them up, you might have to rebind them all again (via Global Shortcuts). I haven't used the AUR build so I can't say for sure if this is the case. If you want to be absolutely sure you've removed **everything**, remove `~/.local/share/kwin/scripts/quarter-tiling`, `/usr/share/kwin/scripts/quarter-tiling`, `/usr/share/kservices5/kwin-script-quarter-tiling.desktop` `~/.local/share/kservices5/kwin-script-quarter-tiling.desktop`.

For debugging, I use "KWin Interactive Console", which is not very helpful. Type "wm console" to krunner to find it.

**Comment by ghost** [comment]
Thanks mate! I think at this point, perhaps an Uninstall Script that simply removes all files under those folders would be awesome.

I think one of the issues I have is that I was attempting to add a Gobal Shortcut to toggle gaps (before you did) and I noticed I had missed a single "}" and thus it broke (my code was almost identical to yours), but there is still the Global shortcut there i created.

Do you know where these are housed at all?

**Comment by Jazqa** [comment]
`~/.config/kglobalshortcutsrc`

**Comment by Jazqa** [comment]
Silly me. Plasma has commands for updating and removing a script so you don't have to do it manually.


Install: `plasmapkg --type kwinscript -i path/to/script`
Update: `plasmapkg --type kwinscript -u path/to/script`
Remove: `plasmapkg --type kwinscript -r path/to/script`

If you're inside the script's folder, path/to/script is obviously `.`

**Comment by Jazqa** [comment]
I'll close this as I've added the instructions for uninstalling the script to the readme. I don't think there's a  need to make a shell script for uninstalling as it's a single command (`plasmapkg --type kwinscript -r path/to/quarter-tiling`). Unfortunately, AUR and git not playing nice together is an issue I can't solve as the other one is installing itself in `/usr/share/` and the other one into `~/.local/share`. However, there's no reason to use both. I tried the AUR version with a virtual machine and it seems to function perfectly fine and update itself the moment I push changes to git.
